src/Domain.jl

```diff
@@ -1343,8 +1349,10 @@ end
     C = N/d.V
     P = 1.0e9
     Gs = p[1:length(d.phase.species)]
-    kfs = convert(typeof(y),p[length(d.phase.species)+1:length(d.phase.species)+length(d.phase.reactions)])
-    krevs = convert(typeof(y),getkfkrevs(;phase=d.phase,T=d.T,P=P,C=C,N=N,ns=ns,Gs=Gs,diffs=d.diffusivity,V=d.V,kfs=kfs)[2])
+    kfsnondiff = p[length(d.phase.species)+1:length(d.phase.species)+length(d.phase.reactions)]
+    kfs,krevs = getkfkrevs(;phase=d.phase,T=d.T,P=P,C=C,N=N,ns=ns,Gs=Gs,diffs=d.diffusivity,V=d.V,kfs=kfsnondiff)
```

Comment:
Thanks for the comment!

I think the current code did the same thing as in ConstantTPDomain, i.e. recalculating krevs in each iterations. As you mentioned in our private conversation, recalculating the krevs in each iteration make the code more robust, in case that a user decides to change the parameters during integration using forward diff for jacobian.

I believe that the type change is handled by the next two lines of code that change kfs and krevs to type of y.